### PR TITLE
Update trimming limitations

### LINKED ISSUE #37878: Add: ASP.NET Core and native AOT compatibility
Add info from [ASP.NET Core and native AOT compatibility](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/#asp-net-core-and-native-aot-compatibility#asp-net-core-and-native-aot-compatibility)

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/incompatibilities

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/deploying/trimming/incompatibilities.md

### Document Version Independent Id

44a05400-ca81-d38e-1ce6-b54a9036621e

### Article author

agocke

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #37878

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/deploying/trimming/incompatibilities.md](https://github.com/dotnet/docs/blob/b5f875e7527172f2d8766304e0299b6404e191de/docs/core/deploying/trimming/incompatibilities.md) | [Known trimming incompatibilities](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/trimming/incompatibilities?branch=pr-en-us-37912) |

### CODE REVIEW

File: docs/core/deploying/trimming/incompatibilities.md
Comment (Rick-Anderson) [inline docs/core/deploying/trimming/incompatibilities.md:26]: Should this be under it's own H2?

File: docs/core/deploying/trimming/incompatibilities.md
Comment (eerhardt) [inline docs/core/deploying/trimming/incompatibilities.md:23]: @ericstj @eiriktsarpalis @tarekgh - do we have official docs for the ConfigurationBinder source generator? I'm not sure pointing normal .NET devs to this issues is going to lead them to success in using it.

File: docs/core/deploying/trimming/incompatibilities.md
Comment (eerhardt) [inline docs/core/deploying/trimming/incompatibilities.md:40]: @agocke @sbomer @vitek-karas - I thought I remember seeing something about supporting C++/CLI , but I can't remember. Basically I thought the trimmer would scan the CLI image, but not trim anything inside it.

Comment (vitek-karas) [reply docs/core/deploying/trimming/incompatibilities.md:40]: It depends how the C++/CLI is used. If you call the assembly from managed, it might work (we copy it untouched, and scan it, as you correctly noted). But if you first call it from native, I doubt that will work - we would need the native hosting side of things to kick in, which is not trim compatible (at least not by default).
So I think it's better to simply state that it's not supported... if nothing else, we also don't have any tests for this currently. If we think we should support at least that one direction, then we should add tests for it somewhere. (We do test the trimming part, but I'm pretty sure we don't test it actually working after trimming).

Comment (Rick-Anderson) [reply docs/core/deploying/trimming/incompatibilities.md:40]: I could write.

is not ***CURRENTLY*** supported. It's possible to call an assembly from managed, but that approach is not supported. Calling C++/CLI from native is incompatible with trimming. 

(Emphasis is used to see change, won't be used in the doc).

Or keep as is, the conservative approach. 

I just don't want to stall folks who'd like to eventually migrate to .NET code from not getting started sooner.

File: docs/core/deploying/trimming/incompatibilities.md
Comment (Rick-Anderson) [reply docs/core/deploying/trimming/incompatibilities.md:23]: @layomia _do we have official docs for the ConfigurationBinder source generator?_

File: docs/core/deploying/trimming/incompatibilities.md
Comment (vitek-karas) [reply docs/core/deploying/trimming/incompatibilities.md:40]: Personally I would prefer we say we don't support it (currently). I'm not confident enough to say that we did the work to support it.

**Comment by Rick-Anderson** [comment]
@vitek-karas  @tarekgh I addressed the two remaining concerns, looking for approval.

**Review by tarekgh — APPROVED** [review]
LGTM. 

Will be good to wait for @eerhardt or @vitek-karas approval too as they are the experts in the area.

**Review by eerhardt — APPROVED** [review]
Thank you!

**Pull request merged**
